azurerm/resource_arm_nat_gateway.go

```diff
@@ -0,0 +1,306 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var natGatewayResourceName = "azurerm_nat_gateway"
+
+func resourceArmNatGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmNatGatewayCreateUpdate,
+		Read:   resourceArmNatGatewayRead,
+		Update: resourceArmNatGatewayCreateUpdate,
+		Delete: resourceArmNatGatewayDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"idle_timeout_in_minutes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  4,
+			},
+
+			"public_ip_address_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+			},
+
+			"public_ip_prefix_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+			},
+
+			"resource_guid": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.Standard),
+				}, false),
+				Default: string(network.Standard),
+			},
+
+			"zones": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
```

Comment:
Can we change this to an `Int` and validate that the input is either 1, 2, or 3 with no duplicate `zones`?

Reply:
When I am checking the specs [here](https://github.com/Azure/azure-rest-api-specs/blob/411d4cf447639289e4ee1efa4d066df2773ebccd/specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/natGateway.json#L397), the item of zones is only described as `string`. In other existed resources, some of them use this `azure.SchemaZones()` as schema of field `zones`. I changed this to `azure.SchemaZones()`.